server.js

```diff
@@ -465,6 +374,369 @@ io.on('connection', (socket) => {
+        user: president,
+        message: message
+    };
+    io.sockets.emit('sh-message', report);
+    console.log(president + ': ' + report.message);
+}
+
+function execution() {
+    if (fascists.includes(president)) {
+        console.log("Fascist president. Executing liberal...");
+        target = liberals[Math.floor(Math.random() * (liberals.length))];
+    } else {
+        console.log("Liberal president. Executing a random player...");
+        target = shPlayers[Math.floor(Math.random() * (shPlayers.length))];
+    }
+    killPlayer(target);
+    console.log("Executing " + target);
+    if (target == hitler) {
+        lWins++;
+        io.emit('sh-game-finished', nameOfHitler + " was killed! " + nameOfLiberals + "s win!");
+        gameOver = true;
+    } else {
+        nextRound('');
+    }
+}
+
+//Checks if all votes are in
+function checkVotes() {
+    if (votesAgainstGov + votesForGov == shPlayers.length) {
+        console.log("all votes in");
+        if (votesAgainstGov >= votesForGov) {
+            console.log("voting failed");
+            nextPresident();
+            io.emit('voting-failed', {
+                presNom: presNom,
+                pres: prevPresNom,
+                chan: chanNom,
+                va: votersAgainstGov,
+                vf: votersForGov
+            });
+            votesForGov = 0;
+            votesAgainstGov = 0;
+            rejectedGovs++;
+            votersAgainstGov = [];
+            votersForGov = [];
+            if (rejectedGovs >= 3) {
+                undesirables = [];
+                io.emit('sh-chaos', {
+                    f: fPols,
+                    l: lPols,
+                    t: topThreePolicies[0]
+                });
+                rejectedGovs = 0;
+            }
+            if (!gameOver) {
+                autoChancellor();
+            }
+        } else if (votesForGov > votesAgainstGov) {
+            console.log("voting passed");
+            president = presNom;
+            chancellor = chanNom;
+            undesirables.push(president);
+            undesirables.push(chancellor);
+            io.emit('voting-passed', {
+                pres: president,
+                chan: chancellor,
+                top: topThreePolicies,
+                fPols: fPols,
+                va: votersAgainstGov,
+                vf: votersForGov
+            });
+            votesForGov = 0;
+            votesAgainstGov = 0;
+            rejectedGovs = 0;
+            votersAgainstGov = [];
+            votersForGov = [];
+            if (hitler == chancellor && fPols >= 3 && bots.includes(president)) {
+                fWins++;
+                io.emit('sh-game-finished', nameOfHitler + " was elected after 3 " + nameOfFascists + " policies were enacted. " + nameOfFascists + "s win!");
+                gameOver = true;
+            } else if (bots.includes(chancellor) && !gameOver) {
+                botPres();
+            }
+        }
+    }
+}
```

Comment:
if a vote fails and the game hasn't ended, the next president (if a bot) picks a chancellor.  if it passes, the game checks for a hitler elected victory then triggers the president to pick policies (if a bot)